docs/core/logger.md

```diff
@@ -44,6 +44,8 @@ For a **complete list** of supported environment variables, refer to [this secti
 
 #### Example using AWS Serverless Application Model (SAM)
 
+The `Logger` utility is instantiated outside of the Lambda handler. In doing this, the same instance can be used across multiple invocations inside the same execution environment. This allows `Metrics` to be aware of things like whether or not a given invocation had a cold start or not.
```

Comment:
Thanks for adding this paragraph!

Suggestion #1 (minor): I understand what we are trying to achieve with this paragraph, but I don't know if it's entirely clear what is the advantage of reusing the instance on multiple invocation here. The Metrics reference can also be confusing since the Logger has the cold start logic too. 

Suggestion  #2 (minor): I'd move this text outside of the SAM section as this does not relate to SAM specifically. Maybe it can be formatted as inline block to make it more visible? 
https://squidfunk.github.io/mkdocs-material/reference/admonitions/?h=warning#inline-blocks

Reply:
Makes total sense, I'll open a separate PR after this one is merged as the same "notice" appears also in `Tracer` & `Metrics` which have already been merged.